src/chatd.cpp

```diff
@@ -1487,6 +1500,32 @@ void Chat::onEndCall(karere::Id userid, uint32_t clientid)
     mCallParticipants.erase(key);
 }
 
+void Chat::initChat()
+{
+    mBackwardList.clear();
+    mForwardList.clear();
+    mIdToIndexMap.clear();
+
+    mForwardStart = CHATD_IDX_RANGE_MIDDLE;
+
+    mOldestKnownMsgId = 0;
+    mLastSeenIdx = CHATD_IDX_INVALID;
+    mLastReceivedIdx = CHATD_IDX_INVALID;
+    mNextHistFetchIdx = CHATD_IDX_INVALID;
+    mLastIdReceivedFromServer = 0;
+    mLastIdxReceivedFromServer = CHATD_IDX_INVALID;
+    mLastServerHistFetchCount = 0;
+    mLastHistDecryptCount = 0;
+
+    mLastTextMsg.setState(LastTextMsgState::kNone);
```

Comment:
Maybe it's better to use `LastTextMsgState::clear()` here, so we also clear the `content` and `type`, not only the `state`